test/functional/holds_controller_test.rb

```diff
@@ -33,6 +33,7 @@ class HoldsControllerTest < ActionController::TestCase
   end
 
   test "should update hold successfully" do
+    sign_in @user
```

Comment:
How did this work before without the user signed in?

Reply:
@thisisstephenbetts  
 created pages starting of the page.
 setup do
    @user = holds(:hold1).user
    @hold1 = holds(:hold1)
    @hold2 = holds(:hold2)
    @school_two = schools(:school_two)
  end

Reply:
@thisisstephenbetts  before update method not using this below method
 **@hold.teacher_set.holds_count_for_user(current_user, @hold.id)).**
 I added new functionality to calculate the current-user holds. So added current-user now.

Reply:
Thanks — makes sense.